File: examples/go-app/internal/formatter/formatter.go
package formatter

import (
	"strconv"
	"strings"

	"example.com/go-app/internal/processor"
)

func FormatReport(report processor.Report) string {
	lines := []string{
		"Project: " + report.ProjectName,
		"Lines: " + formatCount(report.LineCount),
		"Filtered:",
	}
	for _, line := range report.FilteredLines {
		lines = append(lines, "- "+line)
	}

	return strings.Join(lines, "\n")
}

func formatCount(value int) string {
	return strconv.Itoa(value)
}

File: examples/rust-app/src/formatter.rs
use crate::processor::Report;

pub fn formatReport(report: Report) -> String {
	let mut lines = vec![
		format!("Project: {}", report.projectName),
		format!("Lines: {}", report.lineCount),
		"Filtered:".to_string(),
	];
	for line in report.filteredLines {
		lines.push(format!("- {}", line));
	}

	lines.join("\n")
}
